pykusto/_src/query.py

```diff
@@ -619,12 +624,28 @@ def by(self, *args: Union[_AssignmentToSingleColumn, BaseColumn, BaseExpression]
             self._by_assignments.append(_AssignmentToSingleColumn(_AnyTypeColumn(column_name), group_exp))
         return self
 
+    def shuffle_key(self, *keys: Union[_AssignmentToSingleColumn, BaseColumn, BaseExpression]):
```

Comment:
I wonder if this function is the best why to add the shuffle keys. why not add it as a parameter in the `__init__()` function? We want to resemble KQL as much as possible. WDYT?

Reply:
It is problematic in terms of backward compatibility.
If we add as an explicit argument before *args - it breaks all the existing summarize calls.
We cannot add an argument with a default value after **kargs.
Which doesn't leave us with much other options

Reply:
Actually, I do see an advantage in defining it explicitly.
I find the original kusto syntax not explicit enough here. 
I like it that way, the same as I think that separating the "summarize" and the "by" commands make sesnse

Reply:
Backward compatibility is a good explanation, by I humbly disagree with you about what would be prettier. this is in fact a parameter of the summarize function, so I believe it should be in the `init` function, if possible.

Reply:
Maybe my last comment wasn't clear enough.
To implement your suggestion we need to change the arguments of `summarize` (query method) not just `_SummarizeQuery`.
And doing that is pretty hard

Reply:
If we add as an explicit argument before *args - it breaks all the existing summarize calls.
We cannot add an argument with a default value after **kargs.
Which doesn't leave us with many other options We can dedicate a specific "karg" for that but it doesn't seem like a better solution for me

Reply:
why not put it between the *args and the **kwargs?